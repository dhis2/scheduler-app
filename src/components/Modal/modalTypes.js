export const DISCARD_FORM = 'DISCARD_FORM'
export const CRON_PRESET = 'CRON_PRESET'
export const DELETE_JOB = 'DELETE_JOB'
export const RUN_JOB = 'RUN_JOB'
