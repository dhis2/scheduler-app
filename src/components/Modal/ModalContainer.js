import React from 'react'
import { connect } from 'react-redux'
import { string, object } from 'prop-types'
import * as rootSelectors from '../../rootSelectors'
import { selectors } from '../../data/modal'
import * as modalTypes from './modalTypes'
import DeleteJobModal from './DeleteJobModal'
import RunJobModal from './RunJobModal'

export const UnconnectedModalContainer = ({ type, props }) => {
    switch (type) {
        case modalTypes.DELETE_JOB:
            return <DeleteJobModal {...props} />
        case modalTypes.RUN_JOB:
            return <RunJobModal {...props} />
        default:
            return null
    }
}

UnconnectedModalContainer.defaultProps = {
    type: null,
    props: {},
}

UnconnectedModalContainer.propTypes = {
    type: string,
    props: object,
}

const mapStateToProps = state => {
    const modal = rootSelectors.getModal(state)

    return {
        type: selectors.getType(modal),
        props: selectors.getProps(modal),
    }
}

export default connect(mapStateToProps)(UnconnectedModalContainer)
