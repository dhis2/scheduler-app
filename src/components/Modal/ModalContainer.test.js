import React from 'react'
import { shallow } from 'enzyme'
import { DumbModalContainer as ModalContainer } from './ModalContainer'
import * as modalTypes from './modalTypes'

describe('<ModalContainer>', () => {
    it('renders nothing when type is null', () => {
        const wrapper = shallow(<ModalContainer type={null} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('renders DeleteJobModal when requested', () => {
        const wrapper = shallow(<ModalContainer type={modalTypes.DELETE_JOB} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('passes props to DeleteJobModal', () => {
        const wrapper = shallow(
            <ModalContainer type={modalTypes.DELETE_JOB} props={{ id: 'id' }} />
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('renders RunJobModal when requested', () => {
        const wrapper = shallow(<ModalContainer type={modalTypes.RUN_JOB} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('passes props to RunJobModal', () => {
        const wrapper = shallow(
            <ModalContainer type={modalTypes.RUN_JOB} props={{ id: 'id' }} />
        )

        expect(wrapper).toMatchSnapshot()
    })
})
