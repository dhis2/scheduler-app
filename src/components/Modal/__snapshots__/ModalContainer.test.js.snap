// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalContainer> passes props to DeleteJobModal 1`] = `
<Connect(DumbDeleteJobModal)
  id="id"
/>
`;

exports[`<ModalContainer> passes props to RunJobModal 1`] = `
<Connect(DumbRunJobModal)
  id="id"
/>
`;

exports[`<ModalContainer> renders DeleteJobModal when requested 1`] = `<Connect(DumbDeleteJobModal) />`;

exports[`<ModalContainer> renders RunJobModal when requested 1`] = `<Connect(DumbRunJobModal) />`;

exports[`<ModalContainer> renders nothing when type is null 1`] = `""`;
